### PR TITLE
Cross target .NET5.0 and use C#9

### PR DESCRIPTION
**Pull request by iamcarbon**
- Cross targets .NET5.0 to eliminate nuget deps (if accepted, we should bump this target to .NET6 in the fall)
- Reacts to nullability changes
- Update xunit and test sdk to latest versions
- Uses target type new()
- Seals Token class
- Updates a few nullability defaults in test / perf projects to reduce warnings
- Eliminate a few culture specific compares
- Update SDK to  5.0.203

### CODE REVIEW

File: LunrCore/LunrCore.csproj [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>netstandard2.0</TargetFramework>
+    <TargetFrameworks>netstandard2.0;net5.0</TargetFrameworks>
     <RootNamespace>Lunr</RootNamespace>
     <Version>2.3.8.5</Version>
```
Comment (bleroy) [inline LunrCore/LunrCore.csproj:6]: Should we bump the version number here?

**Review by bleroy — CHANGES_REQUESTED** [review]
Thanks, this looks great. GitHub seems to be surfacing a few new warnings that look like they've been triggered at least partly by your changes. Can you take a look?

File: LunrCore/LunrCore.csproj (hunk 1 above)
Comment (iamcarbon) [reply LunrCore/LunrCore.csproj]: Bumped to match lunr.js. It looks like your set union fix made that release.

**Comment by iamcarbon** [comment]
@bleroy Had to figure out the CI there. Apologies for the mess isolating that! Ready for review.

**Comment by bleroy** [comment]
No problem, I appreciate the perseverance!

**Review by bleroy — APPROVED** [review]

**Comment by bleroy** [comment]
Thanks for the contribution, love the improvements to the code!

**Pull request merged**